js/boaexchange.js

```diff
@@ -0,0 +1,1021 @@
+        let used = this.safeFloat (balance, 'held_balance', 0);
+        let total = free + used;
+        const account = {
+            'free': free,
+            'used': used,
+            'total': total,
+        };
+        return { 'currency': currency, 'account': account };
+    }
+
+    parseBalances (balances) {
+        let results = { 'info': balances };
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            let row = this.parseBalance (balance);
+            results[row['currency']] = row['account'];
+        }
+        return results;
+    }
+
+    parseCurrencies (currencies) {
+        const result = {};
+        for (let i = 0; i < currencies.length; i++) {
+            const currency = currencies[i];
+            const id = this.safeString (currency, 'id');
+            const code = this.commonCurrencyCode (this.safeString (currency, 'code'));
+            const precision = 8; // default precision, todo: fix "magic constants"
+            const address = undefined;
+            const fee = this.safeFloat (currency, 'tx_fee'); // todo: redesign
+            result[code] = {
+                'id': id,
+                'code': code,
+                'address': address,
+                'info': currency,
+                'type': undefined,
+                'name': currency['name'],
+                'active': currency['maintenance'],
+                'fee': fee,
+                'precision': precision,
+                'limits': {
+                    'amount': {
+                        'min': Math.pow (10, -precision),
+                        'max': Math.pow (10, precision),
+                    },
+                    'price': {
+                        'min': Math.pow (10, -precision),
+                        'max': Math.pow (10, precision),
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': fee,
+                        'max': Math.pow (10, precision),
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    parseDepositAddress (depositAddress, currency = undefined) {
+        let address = this.safeString (depositAddress['address'], 'address');
+        const code = this.commonCurrencyCode (this.safeString (depositAddress['coin'], 'code'));
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': undefined,
+            'info': depositAddress,
+        };
+    }
+
+    parseFundingLimits (limits, symbol = undefined, params = {}) {
+        return {
+            'info': limits,
+            'min': this.safeFloat (limits['min']),
+            'max': this.safeFloat (limits['max']),
+        };
+    }
+
+    parseLedgerEntries (entries, currency = undefined) {
```

Comment:
Ok, we can also remove this method, because it is declared in the base class as `parseLedger`... Yeah, it doesn't follow the naming system and falls out of it, so, maybe, we will later rename it in the base class. Also, it has to support for `since` and `limit`. We should rename the calls to parseLedgerEntries to parseLedger and respect for the filtering params.

Reply:
so remove this or modify it?

Reply:
I think we shouldn't reinvent it, so, it's safe to remove it and use the base-method instead.

Reply:
done